plugin-copilot/src/main/java/appland/copilotChat/CopilotStartupNotificationActivity.java

```diff
@@ -40,5 +41,9 @@ public void runActivity(@NotNull Project project) {
                 ApplicationManager.getApplication().invokeLater(AppMapNotifications::showFirstCopilotIntegrationEnabled);
             }
         }
+
+        // Try to download content exclusions.
+        // Note if it fails it will be retried later.
+        instance.downloadContentExclusions();
```

Comment:
A `StartupActivity` is invoked when a project is opened.
If 3 projects were restored at startup of the IDE, then the exclusions would be downloaded 3 times.
I'd name this method `ensureDownloadContentExclusions()` and make it it call `ensureContentExclusionsDownloaded(githubToken);` internally to only download the exclusions once.
`GitHubCopilotService` is an appliation-service, i.e. caching the state is fine.

Reply:
I don't think it should be called `ensure` since it's okay-ish if it fails. I'll add a check there to make sure it doesn't redownload it.

Reply:
Ok, I reworked it a bit, let me know if you think it's better.